packages/blockchain-wallet-v4-frontend/src/scenes/plugin/Activity/TransactionList/TransactionList.tsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+import { connect } from 'react-redux'
+import { isEmpty } from 'ramda'
+import styled from 'styled-components'
+
+import { EthProcessedTxType } from '@core/transactions/types'
+import { Text } from 'blockchain-info-components'
+import { Flex } from 'components/Flex'
+import { Padding } from 'components/Padding'
+
+import { getBlockHeight } from '../../../Transactions/NonCustodialTx/Confirmations/selectors'
+import FullScreenModal from '../../common/FullScreenModal'
+import { getIsConfirmed } from '../statusUtils'
+import TransactionDetails from '../TransactionDetails'
+import TransactionListItem from './TransactionListItem'
+
+export interface ITransactionListProps {
+  blockHeight: any
+  coin: string
+  transactions: EthProcessedTxType[]
+}
+
+const Wrapper = styled.div`
+  overflow: auto;
+  padding-top: 24px;
+`
+
+const TransactionList: React.FC<ITransactionListProps> = ({ blockHeight, coin, transactions }) => {
+  const [selectedTx, setSelectedTx] = React.useState<EthProcessedTxType | null>(null)
+
+  const confirmedTransactions = transactions.filter((tx) =>
```

Comment:
`useMemo`. Same below. 

In case of rerender after component state update this value will be taken from cache instead of filtering array once again